autometa/config/databases.py

```diff
@@ -66,230 +97,504 @@ class Databases:
+        if self.dryrun or no_update_db or checksums_match:
+            self.config.set('ncbi', 'nr', db_outfpath)
+            logger.debug(f'set ncbi nr: {db_outfpath}')
+            return
+
+        diamond.makedatabase(fasta=db_infpath, database=db_outfpath, nproc=self.nproc)
+        if os.path.basename(db_infpath) == 'nr.gz':
+            # nr.gz will be removed after successful nr.dmnd construction
+            write_checksum(db_infpath, db_infpath_md5)
+            os.remove(db_infpath)
+
         self.config.set('ncbi', 'nr', db_outfpath)
         logger.debug(f'set ncbi nr: {db_outfpath}')
 
     def extract_taxdump(self):
-        """Extract autometa required files from ncbi taxdump directory.
+        """Extract autometa required files from ncbi taxdump.tar.gz archive
+        into ncbi databases directory and update user config with extracted paths.
+
+        This only extracts nodes.dmp, names.dmp and merged.dmp from
+        taxdump.tar.gz if the files do not already exist. If `do_update`
+        was originally supplied as `True` to the Databases instance, then the
+        previous files will be replaced by the new taxdump files.
 
-        Extracts nodes.dmp, names.dmp and merged.dmp from taxdump.tar.gz
+        After successful extraction of the files, a checksum will be written of the
+        archive for future checking and then the archive will be removed to save user disk space.
 
         Returns
         -------
         NoneType
+            Will update `self.config` section `ncbi` with options 'nodes','names','merged'
 
         """
-        taxdump = self.config.get('ncbi', 'taxdump')
+        taxdump_fpath = self.config.get('ncbi','taxdump')
         taxdump_files = [
             ('nodes', 'nodes.dmp'),
             ('names', 'names.dmp'),
             ('merged', 'merged.dmp'),
         ]
-        for option, fname in taxdump_files:
+        for option,fname in taxdump_files:
             outfpath = os.path.join(self.ncbi_dir, fname)
-            if not self.dryrun and not os.path.exists(outfpath):
-                outfpath = untar(taxdump, self.ncbi_dir, fname)
+            if self.dryrun:
+                logger.debug(f'UPDATE (ncbi,{option}): {outfpath}')
+                self.config.set('ncbi', option, outfpath)
+                continue
+            # Only update the taxdump files if the user says to do an update.
+            if self.do_update and os.path.exists(outfpath):
+                os.remove(outfpath)
+            # Only extract the taxdump files if this is not a "dryrun"
+            if not os.path.exists(outfpath):
+                outfpath = untar(taxdump_fpath, self.ncbi_dir, fname)
             logger.debug(f'UPDATE (ncbi,{option}): {outfpath}')
             self.config.set('ncbi', option, outfpath)
+        if self.dryrun:
+            return
+        taxdump_md5 = f'{taxdump_fpath}.md5'
+        write_checksum(taxdump_fpath, taxdump_md5)
```

Comment:
This part is a little confusing to me. I get that making a md5 file of the `taxdump` file before deleting allows you to later check whether the version we have is the same as is found on the FTP server. But why is this not done when the file is downloaded (I mean calculated, to verify the download)? Wouldn't it make sense to also make checksums of the extracted files, so that later we can verify that they are not corrupted?

Reply:
Note: The taxdump.tar.gz download is fairly lightweight. I've also noticed this file gets updated multiple times daily.

Reply:
All files after they are downloaded are checksummed. I'll write checksums out for any of the formatted/extracted files.

See https://github.com/KwanLab/Autometa/pull/77#discussion_r426850961 for details